### PR TITLE
Give errors in reexporting ambiguities

### PR DESCRIPTION
**Pull request by e-kayrakli**
This PR fixes three ambiguities in reexport by making the compiler give an
error.

1. 

```chapel
module A { var x = 2; }
module B { var x = 4; }
module C {
  public import A.x;
}

module Z {
  use C, B;
  proc main() {
    writeln(x);
  }
}
```

  This PR makes the relevant future a test.

2.

```chapel
module A { var x = 3; }
module B { var x = 7; }
module C {
  public import A.x;
  public import B.x;
}
module D {
  import C;

  proc main() {
    writeln(C.x);
  }
}
```

  I bumped into this while working on the first issue. This was untested, this
  PR adds a test.

3. 

```chapel
module A {
  var x = 2;
  var y = 3;
  proc foo() {
    writeln("A.foo");
  }
}
module B {
  var y = 4;
}
module C {
  public import A.{x as y, foo as bar};
}

module Z {
  use C, B;
  proc main() {
    writeln(y);
    bar();
  }
}
```
  Renames in imports are relatively new. This snippet was also compiling where it
  shouldn't. This PR fixes that and adds a test

While there, this also adds two futures for name conflicts in functions that are
not actually conflicts because of the rest of the function signature. That case
doesn't work for reexports with either `use` or `import`.

If we want to move forward with this implementation, I think we need to followup
with a refactor where the multiple versions of `lookupPublicUnqualAccessSym` are
eliminated either through flags, or the whole logic can be merged into
`lookupNameLocally` somehow.

Test:
- [x] local
- [x] gasnet

### CODE REVIEW

**Comment by lydia-duncan** [comment]
I'd find the PR message more helpful if it included the code itself rather than links to the code

File: test/visibility/import/edgeCases/multipleDef.good [hunk 1]
```diff
@@ -1,2 +1,2 @@
-multipleDef.chpl:9: error: symbol x is multiply defined
+multipleDef.chpl:2: error: symbol x is multiply defined
```
Comment (lydia-duncan) [inline test/visibility/import/edgeCases/multipleDef.good]: The line number here (and below) should use that of the imports, since the imports are what is causing the conflict.  This would be solved by the map you avoided adding for the change.  I would be okay with tackling that in another commit, though

File: test/visibility/import/edgeCases/multipleDef2.good [hunk 2]
```diff
@@ -0,0 +1,4 @@
+multipleDef2.chpl:2: error: symbol x is multiply defined
+multipleDef2.chpl:5: note: also defined here
+multipleDef2.chpl:14: In function 'main':
+multipleDef2.chpl:15: error: Symbol 'x' undeclared in module 'C'
```
Comment (lydia-duncan) [inline test/visibility/import/edgeCases/multipleDef2.good]: Yeah, I don't really like the multiple errors, the presence of second one seems like it could confuse the user

Comment (lydia-duncan) [reply test/visibility/import/edgeCases/multipleDef2.good]: Is the additional error for multipleDef3 indicating that we run down the same path multiple times?

Comment (e-kayrakli) [reply test/visibility/import/edgeCases/multipleDef2.good]: I think not: We run through `public import`s in `C` and see that there are conflicting symbols, give first two lines of error, don't return any one of the symbols. As a result `C.x` doesn't resolve, and we give the other two lines of error.

File: test/visibility/import/edgeCases/multipleDef3.good [hunk 3]
```diff
@@ -1,2 +1,4 @@
 multipleDef3.chpl:2: error: symbol x is multiply defined
 multipleDef3.chpl:5: note: also defined here
+multipleDef3.chpl:2: error: symbol x is multiply defined
+multipleDef3.chpl:5: note: also defined here
```
Comment (e-kayrakli) [inline test/visibility/import/edgeCases/multipleDef3.good]: This multiple error case is a bit different, and I'll spend some time on trying to remove this.

File: test/visibility/import/edgeCases/multipleDef2.good (hunk 2 above)
Comment (e-kayrakli) [reply test/visibility/import/edgeCases/multipleDef2.good]: Among the multiple error cases, I think I like this one and dislike the other one. But wondering what you think as well

Comment (lydia-duncan) [reply test/visibility/import/edgeCases/multipleDef2.good]: I don't really like either, since technically there is a symbol 'x' defined in C, just multiply defined.  But I definitely like this better than the other one

File: compiler/passes/scopeResolve.cpp [hunk 4]
```diff
@@ -1478,20 +1479,27 @@ static void resolveModuleCall(CallExpr* call) {
           }
 
         } else {
-          if (!uSE || uSE->unresolved == mod->name) {
-            USR_FATAL_CONT(call,
-                           "Symbol '%s' undeclared in module '%s'",
-                           mbrName,
-                           mod->name);
+          // we didn't get a symbol, but it maybe due to multiply defined
```
Comment (lydia-duncan) [inline compiler/passes/scopeResolve.cpp]: Minor nit: "may be" is different than "maybe"
